src/redis-check-aof.c

```diff
@@ -51,45 +67,145 @@ int consumeNewline(char *buf) {
     return 1;
 }
 
-int readAnnotations(FILE *fp) {
-    char buf[AOF_ANNOTATION_LINE_MAX_LEN];
+#define AOF_MANIFEST_KEY_FILE_NAME   "file"
+#define AOF_MANIFEST_KEY_FILE_SEQ    "seq"
+#define AOF_MANIFEST_KEY_FILE_TYPE   "type"
+#define AOF_MANIFEST_MAX_LINE        1024
+static aofManifest *loadManifestFromFile(char *am_filepath) {
```

Comment:
why did you have to copy all that parsing logic here?
let's find a way to re-use the one in aof.c

it would be wrong for us to have two copies of this code.
e.g. a bug is fixed in one and left in the other.

we do already have such cases with redis-check-rdb and redis-check-aof.. let's at least avoid adding more of them.

Reply:
Indeed, I am also struggling with this issue. In `aof.c`, the `aofLoadManifestFromDisk` function has no parameters, and it will automatically construct the real path of the manifest according to `server.aof_dirname `and `server.aof_filename`. Then, in `redis-check-aof`, we will directly specify the path of the manifest file, which prevents me from directly reusing the `aofLoadManifestFromDisk` code.

two options:
1.Parse the incoming manifest path, and set the path part and the file name part to `server.aof_dirname` and `server.aof_filename`, which may be a pity, but I think it's a bit complicated.   
2.Refactor the `loadManifestFromFile` function in `aof.c` , split it into two functions `aofLoadManifest(void)` and `aofLoadManifestFromFile(char *filename)`, `aofLoadManifest` internally constructs the path based on `server.aof_dirname` and `server.aof_filename` and passes it to `aofLoadManifestFromFile`, so that the `aofLoadManifestFromFile` function can also be directly reused in `aof-check-aof.c`.

i prefer 2, @oranagra WDYT?

Reply:
Already updated PR as per 2